src/sampler.jl

```diff
@@ -8,14 +8,19 @@ function reconstruct(
     return reconstruct(h, metric=metric)
 end
 
-reconstruct(τ::AbstractProposal, ::AbstractAdaptor) = τ
+reconstruct(κ::AbstractProposal, ::AbstractAdaptor) = κ
```

Comment:
Maybe: `AbstractProposal` ==> `AbstractKernel`? This can be done in a separate PR. Also, there might be a similar type in `AdvancedMH` which we can inherit here.

Reply:
Yes I want to do this change as well but just as you said I want to avoid unnecessary changes for this PR.